src/lib/pickles/step.ml

```diff
@@ -179,8 +179,23 @@ struct
           }
         in
         let env =
+          let module Env_bool = struct
+            type t = bool
+
+            let true_ = true
+
+            let false_ = false
+          end in
```

Comment:
2 minor remarks:
1. This module definition does not depend on the function's argument and thus should be at the same level as the function or higher to be "statically" defined.
2. since this is the same `Env_bool` as for the other `Env_field`, this could (should?) be factored out and defined in another module (this is at best another PR)